apps/api/src/controllers/competition.controller.ts

```diff
@@ -461,15 +461,25 @@ export function makeCompetitionController(services: ServiceRegistry) {
           ? CompetitionStatusSchema.parse(req.query.status)
           : undefined;
         const pagingParams = PagingParamsSchema.parse(req.query);
-        const competitions = await services.competitionManager.getCompetitions(
-          status,
-          pagingParams,
-        );
+        const { competitions, total } =
+          await services.competitionManager.getCompetitions(
+            status,
+            pagingParams,
+          );
 
-        // Return the competitions
+        // Calculate hasMore based on total and current page
+        const hasMore = pagingParams.offset + pagingParams.limit < total;
+
+        // Return the competitions with metadata
         res.status(200).json({
           success: true,
           competitions: competitions,
+          metadata: {
```

Comment:
my fault for initially saying this should be `metadata`, but we want to call it `pagination`

Reply:
> I think the spec doc calls it metadata. Maybe change to spec to pagination too?

agreed. some chats about this from the morning [here](https://recalllabs.slack.com/archives/C08NS37EMDM/p1748877446235329); `pagination` feels more accurate, anyways